tomviz/ModuleManager.cxx

```diff
@@ -232,6 +242,53 @@ bool ModuleManager::serialize(pugi::xml_node& ns, const QDir& saveDir) const
       std::string(TOMVIZ_VERSION_EXTRA).c_str());
   }
 
+  // Iterate over all data sources and check is there are any that are not
+  // currently saved.
+  int modified = 0;
+  foreach (const QPointer<DataSource>& ds, this->Internals->DataSources) {
+    if (ds != nullptr &&
+        ds->persistenceState() == DataSource::PersistenceState::Modified) {
+      modified++;
+    }
+  }
+
+  foreach (const QPointer<DataSource>& ds, this->Internals->ChildDataSources) {
+    if (ds != nullptr &&
+        ds->persistenceState() == DataSource::PersistenceState::Modified) {
+      modified++;
+    }
+  }
+
+  if (modified > 0) {
+
+    QMessageBox modifiedMessageBox;
+    modifiedMessageBox.setIcon(QMessageBox::Warning);
+    QString text = "The pipeline contains an unsaved data source.";
+    QString infoText = "The unsaved data source is shown in the pipeline in "
+                       "italics with an asterisk. "
+                       "You may continue to save that state and this data "
```

Comment:
I think I would tweak the text a little,

    text = "Warning: unsaved data - %1 data source";

then if > 1 append an 's' and it will keep me happy :-)

    infoText = "Unsaved data is marked in the pipeline italic text with an asterisk. You may continue to save the state, and any unsaved data (along with operators/modules) will be skipped.";